WordPress/gutenberg
# Framework: Deprecate `uid` in favor of `clientId`

Related: #7669

This pull request seeks to deprecate all references to "UID" in the codebase with equivalent "Client ID" naming. This change is being made in an effort to promote consistency on abbreviation capitalization, where, for example, previous selectors...

- `getAdjacentBlockUid`
- `getBlockRootUID`

...become:

- `getAdjacentBlockClientId`
- `getBlockRootClientId`

Observing the changes, you should see a number of other inconsistencies having been addressed here, including references to a block's `id` where `uid` is intended.

**Note:** This is a massive pull request, which is prone to becoming stale very quickly. I'd urge that this be acted upon soon to avoid issues with merge conflicts.

**Implementation notes:**

This does include deprecations, not only for the selectors, but also covering components which we expose on `wp.editor` which accept some prop referencing a "uid". This is achieved through a new (and temporary) `withDeprecatedUniqueId` higher-order component.

**Testing instructions:**

Ensure all forms of automated tests pass.

Verify there are no lingering references to UID in the entire codebase, aside from those necessary for temporary compatibility until total removal.

Ensure that deprecations work effectively. Here's a test block as an example, whose `edit` function depends on both the `id` prop being passed and the `BlockTitle` component accepting a `uid` prop:

```js
wp.blocks.registerBlockType( 'aduth/test-uid', {
	title: 'Test UID Block',

    category: 'common',
    
    edit: function( props ) {
        return wp.element.createElement( wp.editor.BlockTitle, { uid: props.id } );
    },
    
    save: function() {
        return null;
    }
} );
```
